tests/shell/shell_recorder/README.md

```diff
@@ -31,279 +34,38 @@ user/test/key1
 user/test/key2
 ```
 
-You can check this exact output in a shell recorder script via the following code:
+. You can check this exact output in a Shell Recorder script via the following code:
 
 ```
 STDOUT: user/test/key1⏎user/test/key2
 < ls user/test
 ```
 
-If you only want to check that `key1` and `key2` are part of the output you can use the regex `key1.*key2` instead:
+. If you only want to check that `key1` and `key2` are part of the output you can use the regex `key1.*key2` instead:
 
 ```
 STDOUT-REGEX: key1.*key2
 < ls user/test
 ```
 
-As you can see the line ending is considered  a normal character (`.`) in the output.
+. As you can see the line ending is considered  a normal character (`.`) in the output.
 
-Options:
+### Options
 
-* STDOUT:  		matched 1:1 against the command output
-* STDOUT-REGEX:		uses posix extended regex
-* STDERR:
-* RET:
-* WARNINGS:
-* ERRORS:
-* DIFF:
+The Shell Recorder provides the following checks
 
-## Commands
+- `STDOUT:` The Shell Recorder matches the **text** after this directive 1:1 with the standard output of the command.
+- `STDOUT-REGEX:` Use this directive if you want to compare the standard output of the command with a **regular expression**.
+* `STDERR:` The Shell Recorder compares the **regular expression** after this directive with the standard error output of the command.
+* `RET:` The Shell Recorder compares this **regular expression** with the return code (exit status) of the command.
+* `WARNINGS:` This **comma separated list** of numbers is compared with the warnings thrown by a `kdb` command.
+* `ERRORS:` The Shell Recorder compares this **comma separated list** of numbers with the errors thrown by a `kdb` command.
+* `DIFF:` This **regular expression** is compared with the output of a `diff` command. The input for the `diff` command is the last and current state of the key database. The directive `DiffType` specifies if the the Shell Recorder compares the file of the storage plugin (`File`) directly, or if it uses the INI (`Ini`) or Dump (`Dump`) format.
```

Comment:
As discussed is this hardly useful as-is, especially because the format is not clearly specified. If someone has a better idea how to specify the expected changes in KDB, please create an issue to describe a feature that allows us to assert KDB content.